include/open62541pp/types/Variant.h

```diff
@@ -457,9 +460,15 @@ void Variant::setArrayImpl(
 }
 
 template <typename T>
-void Variant::setScalarCopyImpl(const T& value, const UA_DataType& dataType) {
-    auto native = detail::allocateUniquePtr<T>(dataType);
-    *native = detail::copy(value, dataType);
+void Variant::setScalarCopyImpl(T&& value, const UA_DataType& dataType) {
+    using ValueType = std::remove_cv_t<std::remove_reference_t<T>>;
+    auto native = detail::allocateUniquePtr<ValueType>(dataType);
+    // https://stackoverflow.com/a/50086839/9967707
+    if constexpr (std::is_lvalue_reference_v<T>) {
+        *native = detail::copy(value, dataType);
+    } else {
+        *native = std::exchange(value, {});
```

Comment:
can't we just move assign here?

Reply:
Integral values and simple C `struct`s will just copy if using `std::move`. Zero-initializing the source object prevents maybe a necessary deallocation of the source object.